core/controllers/profile.py

```diff
@@ -480,6 +485,21 @@ def post(self):
 class SiteLanguageHandler(base.BaseHandler):
     """Changes the preferred system language in the user's preferences."""
 
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'PUT': {
+            'site_language_code': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_regex_matched',
+                        'regex_pattern': '^[a-zA-Z]{2}$'
```

Comment:
Can you check whether just lowercase/uppercase would be enough?

Reply:
@vojtechjelinek Just updated the regex pattern so that it can detect language formats in "en", "en-US", "zh-Hant-TW", "En-au". 

Tested locally and the regex was able to identify, attached is a proof that it correctly triggered when being caled site-language-handler
![Screen Shot 2021-10-04 at 11 19 12 AM](https://user-images.githubusercontent.com/72321026/135887869-4baf5017-1436-4ebd-810e-06357f51e1c3.png)

Reply:
@vojtechjelinek I also tested the langue code that are in asset/constants.ts and they work accordingly

Reply:
Thanks! Great job.